pkg/distributor/distributor.go

```diff
@@ -641,7 +641,7 @@ func (d *Distributor) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReq
 			}
 
 			if errors.Is(err, tooManyClustersError{}) {
-				validation.DiscardedSamples.WithLabelValues(validation.TooManyHAClusters, userID).Add(float64(numSamples))
+				validation.DiscardedSamples.WithLabelValues(validation.ReasonTooManyHAClusters, userID).Add(float64(numSamples))
 				return nil, httpgrpc.Errorf(http.StatusBadRequest, "%s", err)
```

Comment:
I think this can be simplified. Please double check it.

```suggestion
				return nil, httpgrpc.Errorf(http.StatusBadRequest, err.Error())
```